tests/vm_test.go

```diff
@@ -312,6 +312,27 @@ var _ = Describe("[rfe_id:1177][crit:medium][vendor:cnv-qe@redhat.com][level:com
 			return updatedVM
 		}
 
+		validateGenerationState := func(vm *v1.VirtualMachine, expectedGeneration int, expectedDesiredGeneration int, expectedObservedGeneration int, expectedGenerationAnnotation int) {
+			By("By validating the generation states")
+			Eventually(func(g Gomega) {
+				vm, err = virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				g.Expect(err).ToNot(HaveOccurred())
+
+				By("Expecting the generation to match")
+				g.Expect(vm.Generation).To(Equal(int64(expectedGeneration)))
+
+				By("Expecting the generation state in the vm status to match")
+				g.Expect(vm.Status.DesiredGeneration).To(Equal(int64(expectedDesiredGeneration)))
+				g.Expect(vm.Status.ObservedGeneration).To(Equal(int64(expectedObservedGeneration)))
+
+				By("Expecting the generation annotation on the vmi to match")
+				vmi, err := virtClient.VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &k8smetav1.GetOptions{})
+				g.Expect(err).ToNot(HaveOccurred())
+
+				g.Expect(vmi.Annotations).Should(HaveKeyWithValue(v1.VirtualMachineGenerationAnnotation, fmt.Sprintf("%v", expectedGenerationAnnotation)))
```

Comment:
I have never seen such nesting. From my perspective we are telling ginkgo here the following:

repeatedly try to verify that this function succeeds, while at the same time saying with the `g.*` matchers in this function: If you hit an error here, abort the whole test right away.

Reply:
Never mind. Done correctly. It is passed in in the function context.